packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java

```diff
@@ -142,28 +140,20 @@ public void onReceive(Context context, Intent intent) {
     }
   }
 
-  private Task<Void> deleteToken(Map<String, Object> arguments) {
+  private Task<Void> deleteToken() {
     return Tasks.call(
         cachedThreadPool,
         () -> {
-          String senderId =
-              arguments.get("senderId") != null
-                  ? (String) arguments.get("senderId")
-                  : Metadata.getDefaultSenderId(FirebaseApp.getInstance());
-          FirebaseInstanceId.getInstance().deleteToken(senderId, "*");
+          Tasks.await(FirebaseMessaging.getInstance().deleteToken());
           return null;
         });
   }
 
-  private Task<Map<String, Object>> getToken(Map<String, Object> arguments) {
+  private Task<Map<String, Object>> getToken() {
     return Tasks.call(
         cachedThreadPool,
         () -> {
-          String senderId =
-              arguments.get("senderId") != null
-                  ? (String) arguments.get("senderId")
-                  : Metadata.getDefaultSenderId(FirebaseApp.getInstance());
-          String token = FirebaseInstanceId.getInstance().getToken(senderId, "*");
+          String token = Tasks.await(FirebaseMessaging.getInstance().getToken());
```

Comment:
This is potentially a big deal - needs testing to determine if deleteToken/getToken actually behave now, c.f. 

https://github.com/firebase/firebase-ios-sdk/issues/6433#issuecomment-693502927
https://github.com/invertase/react-native-firebase/issues/3714#issuecomment-721027986
https://github.com/invertase/react-native-firebase/issues/1510#issuecomment-717514608

Reply:
For anyone checking this later, if that ever happens. I added a test for this scenario in react-native-firebase that exercised the underlying ios / android SDKs and their behavior appears correct now, this is not an issue (anymore) as far as I can tell. :+1: